src/models/media.ts

```diff
@@ -0,0 +1,32 @@
+import type { SupportedMediaType } from '~/constants/media'
+
+export interface Tag {
+  name: string
+}
+
+export interface ApiMedia {
+  id: string
+  title: string
+
+  creator: string
+  creator_url: string
+
+  url: string
+  foreign_landing_url: string
+
+  license: string
+  license_version: string
+  license_url: string
+
+  tags?: Tag[]
+}
+
+/**
+ * Stores properties common to all media items. This is extended by interfaces
```

Comment:
Can we add the extensions for the individual media types here, too?

Reply:
That will significantly enlarge/delay this PR. Can I do that separately?

Reply:
I'll also need a better understanding because the API uses the same serialiser for for search/result and data but with some `required=False` fields so the search output lacks fields in the search results.

Ideally, we can split the serializer with into two so that we don't use `required=False` fields. That'll give us 1-1 correspondence in the serializers and the media interfaces.